### PR TITLE
HT-54 : Updated the `AppModel` for authentication services

### PR DESCRIPTION
**Pull request by ancoreraj**

### CODE REVIEW

File: packages/api/source/models/AppModel.ts [hunk 1]
```diff
@@ -4,6 +4,35 @@ import paginate from "mongoose-paginate-v2";
...
+            type: Boolean,
+            required: true,
+        },
+        clientId: {
+            type: String,
+            required: true,
+        },
+        secret: {
+            type: String,
+            required: true,
```
Comment (joelerego) [inline packages/api/source/models/AppModel.ts:19]: You can remove the `required` from all three. The Joi schema in the controllers don't have required.

File: packages/api/source/graphql/privateAPI.ts [hunk 2]
```diff
@@ -98,6 +98,16 @@ const typeDefs = gql`
         key: String!
     }
 
+    input GoogleAuthInput {
+        enabled: Boolean!
+        clientId: String!
+        secret: String!
```
Comment (joelerego) [inline packages/api/source/graphql/privateAPI.ts:104]: All three aren't required. Remove the `!` please.

**Review by joelerego — CHANGES_REQUESTED** [review]

File: packages/api/source/models/AppModel.ts (hunk 1 above)
Comment (ancoreraj) [reply packages/api/source/models/AppModel.ts:19]: But it is supposed to be required according to [jira ticket](https://hypertool.atlassian.net/browse/HT-54)

Comment (ancoreraj) [reply packages/api/source/models/AppModel.ts:19]: So we have to add `required` to Joi schema then?

**Review by joelerego — APPROVED** [review]

**Pull request merged**